pkg/yum/repository.go

```diff
@@ -206,8 +207,10 @@ func (r *Repository) Comps() (*Comps, int, error) {
 
 	defer resp.Body.Close()
 
-	if comps, err = ParseCompsXML(resp.Body); err != nil {
-		return nil, resp.StatusCode, fmt.Errorf("error parsing comps.xml: %w", err)
+	if strings.Contains(compsURL, "comps") {
```

Comment:
I don't think that the url actually has to have comps in the name (even if it generally is).  I'd probably adjust this line: https://github.com/content-services/yummy/blob/master/pkg/yum/repository.go#L341-L343  to simply return "", with no error.  And then here, you can check  `if compsURL != ""`

Reply:
oh i see, you removed that line below!

Reply:
but i think you still want to just return empty string

Reply:
@jlsherrill i'm checking for the string "comps" because the getCompsURL method still returns the base URL so it won't be empty. i tried checking that it was empty first then realized that. do you mean i should return an empty string from the getCompsURL method?

Reply:
just pushed changes. is this what you were thinking? had to check that the comps url was empty in multiple places

Reply:
yep! that looks good, since comps is optional, it makes sense to check the return value anytime you need it to do something with it.   A couple of alternatives:

*  change it to return a pointer, and change the empty string check to a nil check.  I tend to like this more, but from what i've read you shouldn't necessarily use a pointer for just this reason (I could be wrong though)
* have it return an empty string with a custom error and then explicitly check for that error

Reply:
hmm i do like the first alternative as well since that would be more consistent with what you all are doing elsewhere, though i'm not sure how to change the URL type to return a pointer to a string without adding a custom URL type. 

```
url.Path = path.Join(url.Path, compsLocation)
	return url.String(), nil
```

Reply:
i guess i can change the return type for getCompsURL to be a pointer to a url.URL and then in Comps() in the Get() i can pass in compsURL.String()

Reply:
traditionally you'd have to store it as a variable, and then reference a pointer:
```
urlString := url.String()
return &urlString, nil
```

We use a library called pointy that can help out though:

```
return pointy.Pointer(url.String()), nil
```

Reply:
oh i see!

Reply:
updated that with my latest push